### DISCUSSION TITLE
How to get disk_storage_available by Java sdk

### DISCUSSION THREAD

**Discussion by Andy6Loda** [minio/minio#11386] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
1. Can i set bucket_storage_total like mc-client 'mc admin bucket quota'
2. Can i get those value(disk_storage_total, ..._used, ..._available) by Java-SDK or  AWS-S3, without prometheu?
3  The mc-client "admin info" just get the disk_storage_tota, but i also need those of every bucket

**Describe the solution you'd like**
1.  I want to know how to get those by Java-SDK or  AWS-S3

**Additional context**
1.mc admin --json info
```
{
    "status": "success",
    "info": {
        "mode": "online",
        "deploymentID": "19c23ac1-a243-4d1c-8add-1f8ffe692b97",
        "buckets": {
            "count": 5
        },
        "objects": {
            "count": 28
        },
        "usage": {
            "size": 4913656
        },
```
2、quota set 
```
[root@test minio]# ./mc admin bucket quota test/dxnout --hard 2GB
Successfully set bucket quota of 1.9 GiB with hard type on `dxnout`
[root@test minio]# ./mc admin bucket quota test/dxnout --hard 2GiB
Successfully set bucket quota of 2.0 GiB with hard type on `dxnout`
```

**Comment by nitisht** [comment]
Please use madmin package https://github.com/minio/minio/blob/master/pkg/madmin/quota-commands.go

**Comment by Andy6Loda** [comment]
@nitisht Thanks for you explain.  But I need work it in Java and c++(aws-sdk-s3) ...

**Comment by Andy6Loda** [comment]
OMG, the titail  is mistake

**Comment by Andy6Loda** [comment]
OKey，Show myself:

```
CopyObjectHandler  + PutObjectHandler + CopyObjectPartHandler + PutObjectPartHandler
↓
enforceBucketQuota
↓
globalBucketQuotaSys.check(){
//1
sys.Get(bucket) 
//2
sys.bucketStorageCache.Get()
↓
dui.BucketsUsage[bucket]
↓
if (bui.Size + uint64(size)) >= q.Quota {
			return BucketQuotaExceeded{Bucket: bucket}
		}
}
```
![image](https://user-images.githubusercontent.com/71309695/104685512-31c86800-5736-11eb-95ba-b6c84618f1b5.png)

Hail the great victory：
GET /minio/admin/v3/datausageinfo

**Comment by Andy6Loda** [comment, marked as answer]
Another way : measure

curl --location --request GET 'http://*****/minio/prometheus/metrics' \
--header 'Authorization: Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjQ3NjQyNzc2OTIsImlzcyI6InByb21ldGhldXMiLCJzdWIiOiJtaW5pb2FkbWluIn0.jLivGZSo9h8nhDrTtrYaoXUw5uWShDzBsY4Mk3iMI0WCHmaZq0hsxpN4UpD0Yc9zXRKhP2KkHfBPoSiYQ7wRMg'

return:    bucket_usage_size{bucket="xxx"} 815990
